### PR TITLE
Update Getting Started doc with Arch Linux specific info

### PR DESCRIPTION
**Pull request by tumido**
Add a section about ManageIQ dependencies, how to install them, and how to configure PostgreSQL.
Add a section about Docker CE installation.

### CODE REVIEW

**Comment by tumido** [comment]
@abonas, would you suggest to leave the common parts for Fedora and Arch (like the Docker setup) at one place and link to it from the Arch section, or would you prefer to duplicate the paragraphs (as it is now in this PR) so it's easier to follow (no need to scroll back and forth) and also more robust to future changes?

**Comment by abonas** [comment]
I think duplication is less preferable because then it is harder to maintain it and it's more work for same result.
but I am open to hear other opinions too. @cfcosta @karelhala @mtho11 @israel-hdez @xeviknal

**Comment by abonas** [comment]
also, if we add more and more envs to the doc it becomes very long. if there's any way to make some sections collapse/expand, that would be great. I don't remember however if it's doable or not in this format.

**Comment by xeviknal** [comment]
From the point of view of the user who is reading the manual, it is easier to have both Fedora/Arch sections completely split. Here an example of [ManageIQ guides](https://github.com/ManageIQ/guides/blob/master/developer_setup.md#install-system-packages)

In order to make it shorter and better readable, we could add an index bullet list with links to specific sections. Somehow like this: [Guides index](https://github.com/ManageiQ/guides).

**Comment by tumido** [comment]
@xeviknal thanks for the examples! The first example was actually the reason why I've chosen to do it this way. On the other hand I have the some concern as @abonas about future updating and the need to cover and pay more attention to duplicates etc..

That being said, collapsible sections are [definitely possible](https://github.com/dear-github/dear-github/issues/166) with Github flavored Markdown. And it would look like this [Gist](https://gist.github.com/tumido/d9d63e4bd5574902133efad13d383d04).

**Comment by israel-hdez** [comment]
I think that having sections for each distro is handy. However, IMO we only should include instructions for distros we can provide support directly and everything else should be on community forums or wikis. Our repo doesn't have a wiki, but we have ManageIQ Talk as the community forum.

**Comment by cfcosta** [comment]
First of all, nicely done! More documentation is always good! That being said, I don't think we should add this to the guide unless we plan on supporting it for a long time. Since most of the people on the team work with Fedora, I think that means that we should support only Fedora and OS X.

If the other people disagree with me, then no problem, let's merge it!

**Comment by abonas** [comment]
@israel-hdez @cfcosta I hear your concerns about supported/unsupported distros, it makes sense, so I agree we can certainly link to it and not make it part of the official guide.

**Comment by cfcosta** [comment]
@abonas what do you mean by "link"? Put it in another markdown file on the repo or link to an external resource? That could be a good blog post or something like that.

Also, I think we should enable wiki on this repository and move all the documentation there.

**Comment by mtho11** [comment]
We are not the first provider to document this kind of stuff, and being under the *ManageIQ org* we should strive to be like the other providers with look-n-feel of documentation. Perhaps under the [ManageIQ Guides](https://github.com/ManageIQ/guides) section is more appropriate and available to a wider audience of ManageIQ users.

IMO the current [Hawkular provider section](https://github.com/ManageIQ/guides/blob/master/providers/hawkular-hawkinit.md) in ManageIQ Guides is just focused on *HawkInit* which should be a sub-section of the more general *Hawkular Provider Setup*.

https://github.com/ManageIQ/guides/blob/master/providers/hawkular-hawkinit.md

To summarize, I think to be consistent with ManageIQ we should:

1. move our current [Hawkular setup Guide](https://github.com/ManageIQ/manageiq-providers-hawkular/blob/master/GETTING_STARTED.md) over to [ManageIQ Guides](https://github.com/ManageIQ/guides). 

Or, 

2. link to our guide from the ManageIQ Guide page -- this might be best because we maintain control over the guide page content merging.

**Comment by abonas** [comment]
> @abonas what do you mean by "link"? Put it in another markdown file on the repo or link to an external resource? That could be a good blog post or something like that.

does not really matter, I think it's easier inside the repo, as we don't have a relevant blog.
>
> Also, I think we should enable wiki on this repository and move all the documentation there.

no objections from my side

**Comment by abonas** [comment]
> We are not the first provider to document this kind of stuff, and being under the ManageIQ org we should strive to be like the other providers with look-n-feel of documentation. Perhaps under the ManageIQ Guides section is more appropriate and available to a wider audience of ManageIQ users.

>
where is the similar documentation for developers for other providers/how does it look? 
 
> IMO the current Hawkular provider section in ManageIQ Guides is just focused on HawkInit which should be a sub-section of the more general Hawkular Provider Setup.
> 
> https://github.com/ManageIQ/guides/blob/master/providers/hawkular-hawkinit.md
> 
> To summarize, I think to be consistent with ManageIQ we should:
> 
> move our current Hawkular setup Guide over to ManageIQ Guides.

then we lose merge rights and the ability to quickly update what we need there, so I would not do that

> Or,
> 
> link to our guide from the ManageIQ Guide page -- this might be best because we maintain control over the guide page content merging.

I am not sure miq core page is the relevant place to link to our very specific provider, and there are many providers in miq, linking to all of them will expand the core dev guide and there's a relatively small team who eventually needs that hawkular guide. 
But you can certainly discuss it with core team first and see if they are interested in adding a link to our guide, if they are - that's great, you can send a PR with the link to our guide.

**Comment by mtho11** [comment]
@abonas in answer to your question:

> where is the similar documentation for developers for other providers/how does it look?

From: https://github.com/ManageIQ/guides

- Provider setup instructions
    - [Amazon AWS](providers/amazon_aws_config.md)
    - [Hawkular](providers/hawkular-hawkinit.md)
    - [Openshift](providers/openshift.md)
    - [Openstack Infra](providers/openstack_infra_provider.md)

**Comment by abonas** [comment]
> @abonas in answer to your question:
> 
> where is the similar documentation for developers for other providers/how does it look?
> From: https://github.com/ManageIQ/guides
> 
> Provider setup instructions
> Amazon AWS
> Hawkular
> Openshift
> Openstack Infra

@mtho11 I see a mismatch between that link that you mentioned and what is published on the manageiq website, it looks different:
I don't see those links in the website and I am unclear what is actually the correct one and mostly used in that case.
http://manageiq.org/docs/guides/developer_setup/

@agrare @blomquisg @jrafanie any idea?

**Comment by jrafanie** [comment]
@abonas Good question.  I don't know.  I wonder if it's manually kept in sync and hasn't been done in a while.  You can ask in https://gitter.im/ManageIQ/manageiq.org.  

It looks like some of that was already brought up here:
https://github.com/ManageIQ/manageiq.org/issues/423

and the getting started guide has been updated, at least in github:
https://github.com/ManageIQ/manageiq.org/pull/588

**Comment by tumido** [comment]
Do we have any resolution on this? Should I rather do a PR against [ManageIQ/guides](https://github.com/ManageIQ/guides) and modify the [developer_setup.md](https://github.com/ManageIQ/guides/blob/master/developer_setup.md)? I'm for unifying the guides or having them bundled somewhere. It seems to me, it's much better accessible and we're not duplicating our efforts. Also keeping all the docs in one repository would be more convenient for maintenance. Is there any possibility of gaining merge rights on [ManageIQ/guides](https://github.com/ManageIQ/guides)?

**Comment by abonas** [comment]
@tumido I  prefer not to place our documentation in the main guide for several reasons:
1. our docs are not relevant for the majority of developers and can confuse them (not everyone should be forking providers-hawkular repo or api repo or starting hawkinit)
2. not having merge rights for our own documentation is very problematic. if we need something urgently updated (or even for onboarding more new developers), it is best to be able to do it quickly and not waiting for merging by another team, which from past experience can be extremely slow.

as noted above in previous comments: as some team members think - because archlinux is unsupported, it should be moved to a separated doc/wiki/whatever is convenient, and  linked from our guide that should remain here in this repo.

**Comment by mtho11** [comment]
@abonas does this mean we should close this at this point?

**Comment by tumido** [comment]
@mtho11 I would say we can close it, I'll write a blog post or a topic in the forum and that's about it. If there's no need to include this in the Hawkular guides, there's no need for having this PR left open/merged. I'll sync with core team on that matter since it's rather a ManageIQ specific guide than a Hawkular one and I'll see. But we can consider this PR closed. ;)
